internal/nginx/config/generator.go

```diff
@@ -28,13 +29,39 @@ type executeFunc func(configuration dataplane.Configuration) []byte
 // It is the responsibility of the caller to validate the configuration before calling this function.
 // In case of invalid configuration, NGINX will fail to reload or could be configured with malicious configuration.
 // To validate, use the validators from the validation package.
-func (g GeneratorImpl) Generate(conf dataplane.Configuration) []byte {
+func (g GeneratorImpl) Generate(conf dataplane.Configuration) []file.File {
+	files := make([]file.File, 0, len(conf.TLSCerts)+1)
+
+	// certs
+	for id, cert := range conf.TLSCerts {
+		contents := make([]byte, 0, len(cert.Cert)+len(cert.Key)+1)
+		contents = append(contents, cert.Cert...)
+		contents = append(contents, '\n')
+		contents = append(contents, cert.Key...)
+
+		files = append(files, file.File{
+			Content:     contents,
+			Path:        generateTLSCertPath(id),
+			Permissions: 0600,
+		})
+	}
+
 	var generated []byte
 	for _, execute := range getExecuteFuncs() {
 		generated = append(generated, execute(conf)...)
 	}
 
-	return generated
+	files = append(files, file.File{
+		Content:     generated,
+		Path:        "/etc/nginx/conf.d/http.conf",
```

Comment:
Let's either make this a const, or what if we updated our dataplane.Configuration to have an `http` section, and in the future we'd add a `stream` section, where each could contain their respective configs and file paths? That's probably out of scope for this, but maybe worth a ticket?